changelog/62676.fixed

```diff
@@ -0,0 +1 @@
+Modified "_get_flags" function so that it returns regex flag names instead of integers
```

Comment:
It doesn't return the names, it returns the flags.
The fix is to stop it failing when flags are provided as input.

Reply:
>The fix is to stop it failing when flags are provided as input.

Ok yes it was failing when ```re.MULTILINE``` was provided an input -- but it was getting that flag name from the ```_get_flags``` output (and was just feeding it back though via the ```_add_flags``` function). But sure we could verify all the input data types on that function. 
I've rarely written tests before, so this is going to be a bit of learning curve for me. I really just need to find a single test where someone's either verified the input and/or the output of an internal function...

Reply:
I reworded this, thanks.